app/models/gids_redis.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'common/models/redis_store'
+require 'common/models/concerns/cache_aside'
+
+# Facade for GIDS.
+class GIDSRedis < Common::RedisStore
+  include Common::CacheAside
+
+  REDIS_CONFIG_KEY = :gi_response
+  redis_config_key REDIS_CONFIG_KEY
+
+  # rest_call [Symbol] the GI::Client method to call
+  # scrubbed_params [Hash] the params to be used with the rest_call
+  attr_accessor :rest_call, :scrubbed_params
+
+  # @return the response returned from GI Client
+  def gi_response
+    @gi_response ||= response_from_redis_or_service
```

Comment:
You definitely don't want to memoize this.
```
service = GIDSRedis.new
service.get_calculator_constants
# => returns calculator constants data
service.get_institution_program_autocomplete_suggestions
# => returns calculator constants data
```

Reply:
removed